File: MyBlogDAL/Entities/Article.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace MyBlogDAL.Entities
{
    public class Article : BaseEntity
    {
        public string Title { get; set; }
        public string Content { get; set; }
        public DateTime DateOfCreation { get; set; }

        public int BlogId { get; set; }
        public virtual Blog Blog { get; set; }
        public int CreatorId { get; set; }
        public virtual User Creator { get; set; }
        public virtual ICollection<Comment> Comments { get; set; }
        public virtual ICollection<Tag> Tags { get; set; }
    }
}

File: MyBlogAngular/src/app/models/article.ts
/*public int Id { get; set; }
        public string Title { get; set; }
        public string Content { get; set; }
        public DateTime DateOfCreation { get; set; }
        public int BlogId { get; set; }
        public string CreatorId { get; set; }
        public string CreatorName { get; set; }*/

export class Article {
  id : number;
  title: string;
  content: string;
  dateOfCreation: Date;
  blogId: number;
  creatorId: string;
  creatorName: string;

  /**
   *
   */
  constructor(id : number,
    title: string,
    content: string,
    dateOfCreation: Date,
    blogId: number,
    creatorId: string,
    creatorName: string) {

    this.id = id;
    this.title = title;
    this.content = content;
    this.dateOfCreation = dateOfCreation;
    this.blogId = blogId;
    this.creatorId = creatorId;
    this.creatorName = creatorName;
  }
}